sentry/src/main/java/io/sentry/SentryTracer.java

```diff
@@ -506,6 +513,25 @@ public void setData(@NotNull String key, @NotNull Object value) {
     return this.root.getData(key);
   }
 
+  @Override
+  public void setMeasurement(@NotNull String name, float value) {
+    if (root.isFinished()) {
+      return;
+    }
+
+    this.measurements.put(name, new MeasurementValue(value, SentryMeasurementUnit.NONE.apiName()));
```

Comment:
You could just forward the call to the other `setMeasurement`, passing `SentryMeasurementUnit.NONE` as 3rd param.

Reply:
On Dart, `unit` is optional, does other SDKs default to `none`?

Reply:
Dunno about other SDKs, but the spec says "defaulting to `none` if missing". https://develop.sentry.dev/sdk/event-payloads/transaction/
![image](https://user-images.githubusercontent.com/4999776/192974297-fd948e9f-d122-47d9-85e7-2896dfd6afe9.png)

I am not sure if they mean that the Relay will set `none` or we should do it on the clients.

Reply:
Seems like both python and js do not default to `none` but rather empty string (which I assume means "optional" in their case)
https://github.com/getsentry/sentry-python/blob/5348834cd6f6b2f877e10febd6ab963166519e04/sentry_sdk/tracing.py#L688

https://github.com/getsentry/sentry-javascript/blob/446a295d3db1b35dd280ba6ee5bdec43c2acbe9e/packages/tracing/src/transaction.ts#L81

Reply:
good catch, please raise this question with the spec and linked repos on client-infra.

Reply:
- In case we decide to keep the default of `NONE` we could call the method with unit from the one without.
- In case we decide to drop the default in the SDK I'd keep them separate and keep the `@NotNull` on the `unit`.